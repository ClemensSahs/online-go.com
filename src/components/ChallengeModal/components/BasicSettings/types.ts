import * as React from 'react';

import {
  OnChangeHandlerInput,
  OnChangeHandlerSelect,
  ChallengeModes,
} from '../types';

export interface BasicSettingsGamePrivateProps {
  dataTestid: string
  checked: boolean
  onChange: OnChangeHandlerInput
};

export interface BasicSettingsGameNameProps {
  gameName: string,
  botId
  botList?
  mode: ChallengeModes
  onChangeChallengeGameName: OnChangeHandlerInput
  onChangeConfBotId: OnChangeHandlerSelect
}

export interface BasicSettingsProps extends BasicSettingsGamePrivateProps,Â BasicSettingsGameNameProps {
  //   gameName: string,
  //   challenge
  //   conf
  //   botId,
  //   botList?,
    gameDemoPrivate
    gamePrivate
  //   mode: ChallengeModes
  //   update_conf_bot_id: OnChangeHandlerSelect
  //   update_challenge_game_name: OnChangeHandlerSelect
    onChangePrivate: OnChangeHandlerInput
    onChangeDemoPrivate: OnChangeHandlerInput
}
