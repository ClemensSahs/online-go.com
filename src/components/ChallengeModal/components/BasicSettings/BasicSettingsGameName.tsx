import * as React from 'react';

import {_, pgettext} from "translate";

import {
  BasicSettingsGameNameProps,
  BotObject,
  ChallengeModes,
} from '../../types';

const BasicSettingsGameName = ({
  gameName,
  botId,
  botList,
  mode,
  onChangeChallengeGameName,
  onChangeConfBotId,
}: BasicSettingsGameNameProps) => {
    if (mode !== ChallengeModes.COMPUTER) {
        return (
            <div className="form-group">
                <label
                  className="control-label"
                  htmlFor="challenge_game_name"
                  >{_("Game Name")}</label>
                <div className="controls">
                    <div className="checkbox">
                        <input
                          type="text"
                          value={gameName}
                          onChange={onChangeChallengeGameName}
                          className="form-control"
                          id="challenge-game-name"
                          placeholder={_("Game Name")}
                        />
                    </div>
                </div>
            </div>
        );
    }

    return (
        <div className="form-group">
            <label
              className="control-label"
              htmlFor="engine"
            >
                {pgettext("Computer opponent", "AI Player")}
            </label>
            <div className="controls">
                <select
                  id="challenge-ai"
                  value={botId}
                  data-testid='challenge-ai'
                  onChange={onChangeConfBotId}
                  required={true}
                >
                    {botList.map((bot: BotObject, idx: string) => {
                        // const rank = rankString(getUserRating(bot).rank);
                        const optionString = `${bot.username} (${bot.rankString})`;
                        return <option key={idx} value={bot.id} >{optionString}</option>;
                    })}
                </select>
            </div>
        </div>
    );
}

export default BasicSettingsGameName
