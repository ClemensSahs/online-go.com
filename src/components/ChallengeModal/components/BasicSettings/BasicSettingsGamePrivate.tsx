import * as React from 'react';

import {_} from "translate";

import {
  BasicSettingsGamePrivateProps,
} from '../../types';

const BasicSettingsGamePrivate = (props: BasicSettingsGamePrivateProps) => {
  return (
      <div className="checkbox">
          <input
            type="checkbox"
            id="challenge-private"
            data-testid={props.dataTestid}
            checked={props.checked}
            onChange={props.onChange}
            />
      </div>
  );
}

export default BasicSettingsGamePrivate
