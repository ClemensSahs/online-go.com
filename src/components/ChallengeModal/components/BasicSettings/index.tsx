export * from './BasicSettings';
export * from './types';
