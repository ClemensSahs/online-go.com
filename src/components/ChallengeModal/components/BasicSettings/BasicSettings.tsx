import * as React from 'react';

import {_} from "translate";

import {
  BasicSettingsProps,
  ChallengeModes
} from '../../types';

import BasicSettingsGameName from './BasicSettingsGameName';
import BasicSettingsGamePrivate from './BasicSettingsGamePrivate';


const getPrivateProps = (props: BasicSettingsProps) => {
  if (props.mode === ChallengeModes.DEMO) {
    return {
      onChange: props.onChangeDemoPrivate,
      checked: props.gameDemoPrivate,
      dataTestid: 'private-input-demo'
    }
  }
  return {
    onChange: props.onChangePrivate,
    checked: props.gamePrivate,
    dataTestid: 'private-input'
  }
}

const BasicSettings = (props: BasicSettingsProps) => {
    const privateProps = getPrivateProps(props);

    return (
    <div
      id="challenge-basic-settings"
      className="left-pane pane form-horizontal"
      role="form"
    >
        <BasicSettingsGameName {...props} />
        <div className="form-group">
            <label className="control-label" htmlFor="challenge-private">
                {_("Private")}
            </label>
            <div className="controls">
                <BasicSettingsGamePrivate {...privateProps} />
            </div>
        </div>
    </div>
  );
}

export default BasicSettings
