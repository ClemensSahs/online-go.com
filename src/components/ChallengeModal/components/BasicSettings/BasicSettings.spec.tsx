
import * as React from 'react';
import { bots_list } from "bots";
import { rankString, getUserRating } from "rank_utils";


import { render, fireEvent, waitFor, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

jest.mock('bots');
jest.mock('rank_utils');

import BasicSettings from './BasicSettings';
import {
  componentSanityChecks,
  runPropsMatrix,
  testCheckBox,
} from 'testUtility';
import {
  ChallengeModes,
} from '../../types';

jest.mock('translate', () => ({
    _: jest.fn(x => x),
    pgettext: jest.fn((x,y) => x),
}));

jest.mock('sockets', () => ({
    termination_socket: {
      on: jest.fn(x => x),
    }
}));

const getBaseProps = (customProps = {}) => {
  const baseProps = {
    botId: null,
    botList: [],
    gamePrivate: false,
    gameDemoPrivate: false,
    gameName: 'test game',
    mode: ChallengeModes.OPEN,
    onChangeChallengeGameName: jest.fn(x => x),
    onChangeConfBotId: jest.fn(x => x),
    onChangeDemoPrivate: jest.fn(x => x),
    onChangePrivate: jest.fn(x => x),
  };
  return {
    ...baseProps,
    ...customProps
  };
}


runPropsMatrix([
  ['normal', getBaseProps()],
  ['as demo', getBaseProps({
    mode: ChallengeModes.DEMO
  })],
  ['as computer', getBaseProps({
    mode:ChallengeModes.COMPUTER,
    botId: 12,
    botList: [
      {id: 'test-bot-1', username: 'test-bot-1', rank: '20', rankString: '10k'},
      {id: 'test-bot-2', username: 'test-bot-2', rank: '21', rankString: '9k'},
      {id: 'test-bot-3', username: 'test-bot-3', rank: '22', rankString: '8k'}
    ],
  })],
  ['as player', getBaseProps({
    mode:ChallengeModes.PLAYER
  })]
],(labelSubfix: string, props) => {
  componentSanityChecks(`BasicSettings.${labelSubfix}`, <BasicSettings {...props} />);
});

describe('ChallengeModes basic settings', () => {

  let mockedProps;
  beforeEach(() => {
    cleanup();
    mockedProps = getBaseProps();
  });


  test('render basic settings check private update', () => {
    testCheckBox({
      Component: BasicSettings,
      mockedProps,
      propertyNameOfValue: 'gamePrivate',
      propertyNameOfUpdateFunction: 'onChangePrivate',
      testid: 'private-input',
    });
  });

  test('render basic settings check private demo update', () => {
    testCheckBox({
      Component: BasicSettings,
      mockedProps: {
        ...mockedProps,
        mode: ChallengeModes.DEMO
      },
      propertyNameOfValue: 'gameDemoPrivate',
      propertyNameOfUpdateFunction: 'onChangeDemoPrivate',
      testid: 'private-input-demo',
    });
  });
});
