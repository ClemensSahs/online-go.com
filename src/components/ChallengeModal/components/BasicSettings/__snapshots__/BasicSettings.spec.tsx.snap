// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componentSanityChecks for BasicSettings.as computer matches the snapshot 1`] = `
<div
  className="left-pane pane form-horizontal"
  id="challenge-basic-settings"
  role="form"
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="engine"
    >
      Computer opponent
    </label>
    <div
      className="controls"
    >
      <select
        data-testid="challenge-ai"
        id="challenge-ai"
        onChange={[MockFunction]}
        required={true}
        value={12}
      >
        <option
          value="test-bot-1"
        >
          test-bot-1 (10k)
        </option>
        <option
          value="test-bot-2"
        >
          test-bot-2 (9k)
        </option>
        <option
          value="test-bot-3"
        >
          test-bot-3 (8k)
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-private"
    >
      Private
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          checked={false}
          data-testid="private-input"
          id="challenge-private"
          onChange={[MockFunction]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`componentSanityChecks for BasicSettings.as demo matches the snapshot 1`] = `
<div
  className="left-pane pane form-horizontal"
  id="challenge-basic-settings"
  role="form"
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge_game_name"
    >
      Game Name
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          className="form-control"
          id="challenge-game-name"
          onChange={[MockFunction]}
          placeholder="Game Name"
          type="text"
          value="test game"
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-private"
    >
      Private
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          checked={false}
          data-testid="private-input-demo"
          id="challenge-private"
          onChange={[MockFunction]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`componentSanityChecks for BasicSettings.as player matches the snapshot 1`] = `
<div
  className="left-pane pane form-horizontal"
  id="challenge-basic-settings"
  role="form"
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge_game_name"
    >
      Game Name
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          className="form-control"
          id="challenge-game-name"
          onChange={[MockFunction]}
          placeholder="Game Name"
          type="text"
          value="test game"
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-private"
    >
      Private
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          checked={false}
          data-testid="private-input"
          id="challenge-private"
          onChange={[MockFunction]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`componentSanityChecks for BasicSettings.normal matches the snapshot 1`] = `
<div
  className="left-pane pane form-horizontal"
  id="challenge-basic-settings"
  role="form"
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge_game_name"
    >
      Game Name
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          className="form-control"
          id="challenge-game-name"
          onChange={[MockFunction]}
          placeholder="Game Name"
          type="text"
          value="test game"
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-private"
    >
      Private
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          checked={false}
          data-testid="private-input"
          id="challenge-private"
          onChange={[MockFunction]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
</div>
`;
